public/app/features/dashboard-scene/panel-edit/PanelVizTypePicker.tsx

```diff
@@ -30,7 +30,21 @@ export function PanelVizTypePicker({ vizManager, data, onChange }: Props) {
   const defaultTab = isWidgetEnabled ? VisualizationSelectPaneTab.Widgets : VisualizationSelectPaneTab.Visualizations;
   const panelModel = useMemo(() => new PanelModelCompatibilityWrapper(panel), [panel]);
 
+  const supportedListModes = useMemo(
+    () =>
+      new Set([
+        VisualizationSelectPaneTab.Widgets,
+        VisualizationSelectPaneTab.Visualizations,
+        VisualizationSelectPaneTab.Suggestions,
+      ]),
+    []
+  );
   const [listMode, setListMode] = useLocalStorage(tabKey, defaultTab);
+  useEffect(() => {
+    if (listMode && !supportedListModes.has(listMode)) {
+      setListMode(defaultTab);
+    }
+  }, [defaultTab, listMode, setListMode, supportedListModes]);
```

Comment:
not sure I understand this, when would Visualizations not be supported ?

Reply:
It's not about Visualizations not being supported, but rather if `listMode` isn't one of `VisualizationSelectPaneTab.Widgets`, `VisualizationSelectPaneTab.Visualizations` or `VisualizationSelectPaneTab.Suggestions`, as in the case where we go from the old arch to scenes having been on the library panels tab in the panel editor (or if for some reason someone manually edited the listMode value to something nonsensical via localstorage).